trino/auth.py

```diff
@@ -546,17 +546,15 @@ def _construct_cache_key(host: Optional[str], user: Optional[str]) -> Optional[s
 
     @staticmethod
     def _parse_authenticate_header(header: str) -> Dict[str, str]:
-        split_challenge = header.split(" ", 1)
-        trimmed_challenge = split_challenge[1] if len(split_challenge) > 1 else ""
+        components = header.split(", ")
         auth_info_headers = {}
 
-        for item in trimmed_challenge.split(","):
-            comps = item.split("=")
-            if len(comps) == 2:
-                key = comps[0].strip(' "')
-                value = comps[1].strip(' "')
-                if key:
-                    auth_info_headers[key.lower()] = value
+        for component in components:
+            if "=" in component:
+                key, value = component.split("=", 1)
+                if value[0] == '"' and value[-1] == '"':
+                    value = value[1:-1]
+                auth_info_headers[key.lower()] = value
         return auth_info_headers
```

Comment:
We could add a bit more exception handling here as well in case the headers don't return the expected values.

Reply:
in the worst case we should make sure enabling DEBUG logs will show the header values (raw values before we try extraction).